frontend/pages/ManageControlsPage/OSUpdates/components/OSVersionTable/OSVersionTable.tsx

```diff
@@ -1,30 +1,92 @@
-import React from "react";
+import React, { useCallback } from "react";
+import { InjectedRouter } from "react-router";
 
+import PATHS from "router/paths";
 import { IOperatingSystemVersion } from "interfaces/operating_system";
+import { getNextLocationPath } from "utilities/helpers";
 
+import { ITableQueryData } from "components/TableContainer/TableContainer";
 import TableContainer from "components/TableContainer";
 
 import { generateTableHeaders } from "./OSVersionTableConfig";
 import OSVersionsEmptyState from "../OSVersionsEmptyState";
+import { parseOSUpdatesCurrentVersionsQueryParams } from "../CurrentVersionSection/CurrentVersionSection";
 
 const baseClass = "os-version-table";
 
 interface IOSVersionTableProps {
+  router: InjectedRouter;
   osVersionData: IOperatingSystemVersion[];
   currentTeamId: number;
   isLoading: boolean;
+  queryParams: ReturnType<typeof parseOSUpdatesCurrentVersionsQueryParams>;
 }
 
-const DEFAULT_SORT_HEADER = "hosts_count";
-const DEFAULT_SORT_DIRECTION = "desc";
-
 const OSVersionTable = ({
+  router,
   osVersionData,
   currentTeamId,
   isLoading,
+  queryParams,
 }: IOSVersionTableProps) => {
   const columns = generateTableHeaders(currentTeamId);
 
+  const determineQueryParamChange = useCallback(
+    (newTableQuery: ITableQueryData) => {
+      const changedEntry = Object.entries(newTableQuery).find(([key, val]) => {
+        switch (key) {
+          case "sortDirection":
+            return val !== queryParams.order_direction;
+          case "sortHeader":
+            return val !== queryParams.order_key;
+          case "pageIndex":
+            return val !== queryParams.page;
+          default:
+            return false;
+        }
+      });
+      return changedEntry?.[0] ?? "";
+    },
+    [queryParams.order_direction, queryParams.order_key, queryParams.page]
+  );
+
+  const generateNewQueryParams = useCallback(
+    (newTableQuery: ITableQueryData, changedParam: string) => {
+      const newQueryParam: Record<string, string | number | undefined> = {
+        team_id: currentTeamId,
+        order_direction: newTableQuery.sortDirection,
+        order_key: newTableQuery.sortHeader,
+        page: changedParam === "pageIndex" ? newTableQuery.pageIndex : 0,
+      };
+
+      return newQueryParam;
+    },
+    [currentTeamId]
+  );
+  // NOTE: this is called once on initial render and every time the query changes
```

Comment:
This issue has been with us for as long as I can remember. Whenever finally we get around to digging into it, I think we might want to look into some of the documentation around debugging with [StrictMode](https://react.dev/reference/react/StrictMode).